sdcm/nemesis.py

```diff
@@ -4205,7 +4205,12 @@ def disrupt_add_remove_mv(self):
         Verify the MV can be used in a query.
         Finally, drop the MV.
         """
-        node1, node2 = self.cluster.nodes[:2]
+        try:
+            node1, node2 = [node for node in self.cluster.nodes if not node.running_nemesis][:2]
```

Comment:
first, we should use `self.target_node` as the node which gets restarted, because now nemesis may take up to 3 nodes to be set as `running_nemesis`. It also will be released later when setting new target node before next nemesis.

The second thing is that still node2 can be left in `running_nemesis` state in case of some uncaught exception along the way. Still this solution is not safe.

Reply:
I agree with @soyacz we should be using target node, and not inventing a new method to select node, node that we only do cql command on doesn't need to be selected as currently running nemesis.

Reply:
> I agree with @soyacz we should be using target node, and not inventing a new method to select node, node that we only do cql command on doesn't need to be selected as currently running nemesis.

@fruch But in that case, won't parallel nemesis cause failure of this one in case the second node gets disrupted?

Reply:
if `cql_connection_patient_exclusive` is used yes, if `cql_connection_patient` is used we connect to all of the up nodes, so it would work regardless

Reply:
to be on the safe side, assuming some code might, the nemesis is marking that node now.

We really should change the API of `cql_connection_patient` to not pass down a node, to avoid this confusion. cause the node isn't really get used, and by default we pass all ip of all nodes

Reply:
regardless of cql-connection, i don't think all non-disruptive nemeses can ran in parallel with this nemesis. For example - `NoCorruptRepairMonkey` would probably fail (with 3 nodes configuration).